### ISSUE TITLE
@cypress/react throws in Node v12

### ISSUE THREAD

**Issue by emilyrohrbough** [cypress-io/cypress#19069]
### Current behavior

Similar to https://github.com/cypress-io/cypress/issues/18973, @cypress/react is throwing an error when using Node 12, cypress v9.0 & @cypress/react v5.10.3.

Found when updating the cypres-example-todomvc-reduc repo to use cypress 9.1.0 & @cypress/react.

```js
yarn component
yarn run v1.22.17
$ cypress open-ct

The function exported by the plugins file threw an error.

We invoked the function exported by `/Users/emily/dev/cypress-example-todomvc-redux/cypress/plugins/index.js`, but it threw an error.

 /Users/emily/dev/cypress-example-todomvc-redux/node_modules/@cypress/react/plugins/utils/get-transpile-folders.js:5
  const rawFolders = config.addTranspiledFolders ?? []
                                                  ^

SyntaxError: Unexpected token '?'
    at Module._compile (internal/modules/cjs/loader.js:891:18)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Users/emily/dev/cypress-example-todomvc-redux/node_modules/@cypress/react/plugins/react-scripts/findReactScriptsWebpackConfig.js:6:33)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Users/emily/dev/cypress-example-todomvc-redux/node_modules/@cypress/react/plugins/react-scripts/index.js:2:39)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
✨  Done in 8.51s.
```

### Desired behavior

@cypress/react is compatible with Node 12. It wouldd also be nice if the package.json defined the node version as `>=12.0.0`.

### Test code to reproduce

https://github.com/cypress-io/cypress-example-todomvc-redux/compare/use-9.1.0

```
nvm use $(cat .node_version)
yarn install
yarn component
```

### Cypress Version

9.1.0

**Comment by yann-combarnous** [comment]
@emilyrohrbough , confirmed, same here using Cypress Github action.

Even after setting

```
      - name: Setup Node
        uses: actions/setup-node@v1
        with:
          node-version: 14
```

as recommended in the documentation, I am getting above error. Had to add 
```
  "nodeVersion": "bundled",
```

in my Cypress.json to workaround the issue.

**Comment by ZachJW34** [comment]
Fixed via #20094 and released as [@cypress/react-v5.12.2](https://www.npmjs.com/package/@cypress/react)
